processing/src/main/java/io/druid/segment/StringDimensionIndexer.java

```diff
@@ -194,19 +222,22 @@ public StringDimensionIndexer(MultiValueHandling multiValueHandling)
     final int oldDictSize = dimLookup.size();
 
     if (dimValues == null) {
-      dimLookup.add(null);
-      encodedDimensionValues = null;
+      if (!dimLookup.contains(null)) {
+        encodedDimensionValues = new int[]{dimLookup.add(null)};
+      } else {
+        encodedDimensionValues = new int[]{dimLookup.getId(null)};
+      }
     } else if (dimValues instanceof List) {
       List<Object> dimValuesList = (List) dimValues;
       if (dimValuesList.isEmpty()) {
         dimLookup.add(null);
         encodedDimensionValues = EMPTY_INT_ARRAY;
       } else if (dimValuesList.size() == 1) {
-        encodedDimensionValues = new int[]{dimLookup.add(STRING_TRANSFORMER.apply(dimValuesList.get(0)))};
+        encodedDimensionValues = new int[]{dimLookup.add(EMPTY_TO_NULL_IF_NEEDED.apply(dimValuesList.get(0)))};
```

Comment:
`EMPTY_TO_NULL_IF_NEEDED` is a function that is held in a static final constant and always called via .apply(). So it could be a static method.

Reply:
done. changed to static method.